### ISSUE TITLE
[ic-dialog] Tabs focus issue

### ISSUE THREAD

**Issue by GCHQ-Developer-530** [mi6/ic-ui-kit#3658]
### Summary of the bug 
When tabs are added to a dialog there are a few issues. The focus gets trapped on the dismiss button and the tabs never receive focus by keyboard navigation unless one of the tabs is clicked. But as soon as you tab away from the tabs then they won't receive focus again.

### 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. Go to [simple Stackblitz example](https://stackblitz.com/edit/ijtmyp7n?file=src%2Fapp.tsx&hideExplorer=1)
2. Open dialog
3. Tab around
4. See the error 

### 🧐 Expected behaviour 
Should be able to tab to the tabs, and also tab away from the dismiss button.

### Additional info 
This fix might also be needed on v2. It's an accessibility issue that keyboard navigation doesn't work and a team has asked for this and they can't migrate to v3. 

The team also has more components in the tab panels themselves e.g. tables/text fields. They're getting inconsistent behaviour when opening the dialog - sometimes focus is on the text field, sometimes focus is on the dismiss button. Would be worth looking into this issue as well.

NOTE: When fixing on V2, if we choose to do it, the code is very different now. Raised as separate ticket #3773
